File: src/main/java/ru/ioleksiv/telegram/bot/core/controller/annotations/parser/filter/primitive/LongFilterParser.java
package ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.primitive;

import org.springframework.stereotype.Component;
import ru.ioleksiv.telegram.bot.api.annotations.filter.primitive.LongFilter;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.filter.FilterParser;
import ru.ioleksiv.telegram.bot.core.controller.annotations.parser.finder.Finder;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.Validator;
import ru.ioleksiv.telegram.bot.core.controller.handler.check.impl.LongValidator;

@Component
public class LongFilterParser implements FilterParser<LongFilter, Long> {

    @Override
    public Validator<Long> createChecker(LongFilter annotation, Finder finder) {
        return new LongValidator(annotation.lessThan(),
                                 annotation.lessOrEqual(),
                                 annotation.moreThan(),
                                 annotation.moreOrEqual());
    }

    @Override
    public Class<LongFilter> getAnnotationClass() {
        return LongFilter.class;
    }

}

File: core/src/main/kotlin/io/github/oleksivio/tl/kbot/core/controller/annotations/parser/filter/primitive/LongFilterParser.kt
package io.github.oleksivio.tl.kbot.core.controller.annotations.parser.filter.primitive

import io.github.oleksivio.tl.kbot.core.annotations.filter.primitive.LongFilter
import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.filter.FilterParser
import io.github.oleksivio.tl.kbot.core.controller.annotations.parser.finder.Finder
import io.github.oleksivio.tl.kbot.core.controller.handler.check.Validator
import io.github.oleksivio.tl.kbot.core.controller.handler.check.impl.LongValidator

class LongFilterParser :
    FilterParser<LongFilter, Long> {

    override val parserAnnotationClass = LongFilter::class

    override fun createChecker(annotation: LongFilter, finder: Finder): Validator<Long> {
        return LongValidator(
            annotation.lessThan,
            annotation.lessOrEqual,
            annotation.greatThan,
            annotation.greatOrEqual
        )
    }
}
